src/praisonai/praisonai/cli/legacy/dispatch/legacy_dispatch.py

```
@@ -0,0 +1,57 @@
+so parse_args can exit once at the top level.
+"""
+
+from __future__ import annotations
+
+import sys
+from typing import TYPE_CHECKING, Callable, Optional
+
+if TYPE_CHECKING:
+    from argparse import Namespace
+
+HandlerFn = Callable[["PraisonAI", "Namespace", list[str]], Optional[int]]
```

Ruff (0.15.20)

[error] Undefined name `PraisonAI`

(F821)